src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AbstractServiceImpl.java

```diff
@@ -55,8 +63,9 @@ public AbstractServiceImpl(MongoRepository<U, String> mongoRepository,
         debugMap.put("company_accounts_id", companyAccountId);
 
         try {
-            baseEntity.setId(generateID(companyAccountId));
+            baseEntity.setId(generateID(companyAccountId, getResourceName()));
             mongoRepository.insert(baseEntity);
+            addParentLink(companyAccountId, selfLink);
```

Comment:
adding the parent link should be done in a separate try-catch block so the error handling (and logging of errors) can be more specific i.e. we know which mongo write failed and handle it appropriately